### PR TITLE
Feature/feature pet profile image

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdateProfileImage.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        trUpdatePetImage.setUser(user);
+        trUpdatePetImage.setPet(pet);
+        trUpdatePetImage.setNewPetImage(newPetImage);
+        trUpdatePetImage.execute();
+
+        assertEquals("Should add pet profileImage", newPetImage, pet.getProfileImage());
+    }
+
+    private Bitmap getBitmap(int color) {
+        Bitmap bitmap = Bitmap.createBitmap(20, 20, Bitmap.Config.ARGB_8888);
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdateProfileImage.java]: Codacy Issue found: ['20' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 13
           

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdateProfileImage.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePetImage.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoom.java  1
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CircularImageView.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641&bid=17125672)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java [hunk 2]
```diff
@@ -15,31 +18,133 @@
...
+            petProfileDrawable = new BitmapDrawable(getResources(), petImage);
+        }
+    }
+
+    /**
+     * Sets the pet profile image drawable.
+     * @param drawable The pet profile drawable to set
+     */
+    public static void setPetProfileDrawable(Drawable drawable) {
+        isImageModified = isImageModified || drawable != petProfileDrawable;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java]: Codacy Issue found: [Use equals() to compare object references.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java [hunk 3]
```diff
@@ -15,31 +18,133 @@
...
+    /**
+     * Sets the pet profile image.
+     */
+    private void setPetProfileImage() {
+       petProfileImage = binding.imgPet;
+
+       if (petProfileDrawable == null) {
+           assignPetImageToDisplay();
+           isImageModified = false;
+       }
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java]: Codacy Issue found: ['if rcurly' has incorrect indentation level 7, expected level should be 8.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java [hunk 4]
```diff
@@ -15,31 +18,133 @@
...
+        } else {
+            binding.inputGender.setText(R.string.female);
+        }
+    }
+
+    /**
+     * Sets the pet profile image.
+     */
+    private void setPetProfileImage() {
+       petProfileImage = binding.imgPet;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java]: Codacy Issue found: ['method def' child has incorrect indentation level 7, expected level should be 8.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/NotImplementedFragment.java [hunk 5]
```diff
@@ -9,10 +9,13 @@
 import android.view.ViewGroup;
 
 import org.pesmypetcare.mypetcare.R;
+import org.pesmypetcare.mypetcare.activities.views.CircularImageView;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/NotImplementedFragment.java]: Codacy Issue found: [Unused import - org.pesmypetcare.mypetcare.activities.views.CircularImageView.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java [hunk 6]
```diff
@@ -15,31 +18,133 @@
...
+
+       if (petProfileDrawable == null) {
+           assignPetImageToDisplay();
+           isImageModified = false;
+       }
+
+       petProfileImage.setDrawable(petProfileDrawable);
+
+       petProfileImage.setOnClickListener(view -> {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java]: Codacy Issue found: ['method def' child has incorrect indentation level 7, expected level should be 8.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoom.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+public class ImageZoom extends Fragment {
+    private static final String[] IMAGE_MIME_TYPES = {"image/jpeg", "image/png"};
+    private static final int GALLERY_ZOOM_REQUEST_CODE = 0;
+    private static final float RADIUS = 1000.0f;
+
+    private FragmentImageZoomBinding binding;
+    private static Drawable drawable;
+
+    public ImageZoom(Drawable drawable) {
+        ImageZoom.drawable = drawable;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoom.java]: Codacy Issue found: [Possible unsafe assignment to a non-final static field in a constructor.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdateProfileImage.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        Canvas canvas = new Canvas(bitmap);
+        Paint paint = new Paint();
+        paint.setColor(color);
+        canvas.drawRect(0F, 0F, 20.0f, 20.0f, paint);
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdateProfileImage.java]: Codacy Issue found: ['20.0f' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java [hunk 9]
```diff
@@ -15,31 +18,133 @@
...
+
+    /**
+     * Sets the pet profile image.
+     */
+    private void setPetProfileImage() {
+       petProfileImage = binding.imgPet;
+
+       if (petProfileDrawable == null) {
+           assignPetImageToDisplay();
+           isImageModified = false;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java]: Codacy Issue found: ['if' child has incorrect indentation level 11, expected level should be 12.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoom.java [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+import java.util.Objects;
+
+public class ImageZoom extends Fragment {
+    private static final String[] IMAGE_MIME_TYPES = {"image/jpeg", "image/png"};
+    private static final int GALLERY_ZOOM_REQUEST_CODE = 0;
+    private static final float RADIUS = 1000.0f;
+
+    private FragmentImageZoomBinding binding;
+    private static Drawable drawable;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoom.java]: Codacy Issue found: [Static variable definition in wrong order.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java [hunk 11]
```diff
@@ -15,31 +18,133 @@
...
+       }
+
+       petProfileImage.setDrawable(petProfileDrawable);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java]: Codacy Issue found: ['method def' child has incorrect indentation level 7, expected level should be 8.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java [hunk 12]
```diff
@@ -15,31 +18,133 @@
...
+    }
+
+    /**
+     * Sets the pet profile image.
+     */
+    private void setPetProfileImage() {
+       petProfileImage = binding.imgPet;
+
+       if (petProfileDrawable == null) {
+           assignPetImageToDisplay();
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java]: Codacy Issue found: ['if' child has incorrect indentation level 11, expected level should be 12.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java [hunk 13]
```diff
@@ -15,31 +18,133 @@
...
+        }
+    }
+
+    /**
+     * Sets the pet profile image.
+     */
+    private void setPetProfileImage() {
+       petProfileImage = binding.imgPet;
+
+       if (petProfileDrawable == null) {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java]: Codacy Issue found: ['if' has incorrect indentation level 7, expected level should be 8.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePetImage.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdateProfileImage.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoom.java  1
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CircularImageView.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641&bid=17125672)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java [hunk 14]
```diff
@@ -251,10 +282,81 @@ public void changeFragmentPass(Fragment fragment) {
...
+        }
+        else {
+            Bitmap bitmap = BitmapFactory.decodeFile(imagePath);
+            Drawable drawable = new BitmapDrawable(getResources(), bitmap);
+            ((ImageZoom) actualFragment).setDrawable(drawable);
+        }
+    }
+
+    /**
+     * Gets the path of the selected image in the gallery
```
Comment (xavier-campos) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Falta el punt en el comentari

Comment (xavier-campos-diaz) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Falta el punt al final del comentari

Comment (AlbertPG) [reply app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Solucionat

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdateProfileImage.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoom.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePetImage.java  3
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CircularImageView.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641&bid=17125672)

**Review by xavier-campos-diaz — APPROVED** [review]

**Review by enrichp — APPROVED** [review]

**Pull request merged**